### PR TITLE
Implement ValidateVolumeCapabilities and refactor parameter handling for more comprehensive validation of existing disks in all cloud calls

### LINKED ISSUE #162: Implement ValidateVolumeCapabilities
Currently this function is unimplemented and just returns a "Unimplemented" message.

This should be done ASAP
/cc @saad-ali

### LINKED ISSUE #384: Validate volume capabilities for file system type and other non-access mode things
/assign

### PR DESCRIPTION
**Pull request by davidz627**
TODO: 
- [x] Add some unit tests for new functions :)

/kind cleanup
/kind feature
/assign @msau42 @misterikkit 

Partially does #384 (Still need to do `Node` side validation in a seperate PR)
Fixes #162 

```release-note
ValidateVolumeCapabilities validates that the given volume conforms to all capabilities in the request. Validation of existing volumes during inserts also improved to check all parameters.
```

### CODE REVIEW

File: pkg/common/parameters.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		DiskEncryptionKMSKey: "",                  // Default
+	}
+	for k, v := range parameters {
+		if k == "csiProvisionerSecretName" || k == "csiProvisionerSecretNamespace" {
+			// These are hardcoded secrets keys required to function but not needed by GCE PD
+			continue
+		}
+		switch strings.ToLower(k) {
+		case ParameterKeyType:
+			if v != "" {
```
Comment (msau42) [inline pkg/common/parameters.go]: should it be considered an error if a user specified a key without a value?

Comment (davidz627) [reply pkg/common/parameters.go]: in-tree PD just treats that case as the default. I tried setting `type: ""` and it just created a `pd-standard` without complaining

**Comment by davidz627** [comment]
#415 seems to have stalled and already has conflicts

**Comment by davidz627** [comment]
@msau42 added unit tests, PTAL

File: pkg/common/parameters.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+				p.DiskType = v
```
Comment (msau42) [inline pkg/common/parameters.go]: do we want toLower() here?

Comment (davidz627) [reply pkg/common/parameters.go]: hmmmm for in-tree we `toLower()` replication type but literally none of the other parameters. It feels like it doesn't matter so much but can definitely be added without issue

**Pull request merged**